File: problems/0011-container-with-most-water/go/solution.go
package solution

// Container With Most Water — https://leetcode.com/problems/container-with-most-water/
func maxArea(height []int) int {
	left, right := 0, len(height)-1
	best := 0
	for left < right {
		h := height[left]
		if height[right] < h {
			h = height[right]
		}
		if area := (right - left) * h; area > best {
			best = area
		}
		if height[left] < height[right] {
			left++
		} else {
			right--
		}
	}
	return best
}

File: problems/0011-container-with-most-water/typescript/solution.ts
// Container With Most Water — https://leetcode.com/problems/container-with-most-water/
export function maxArea(height: number[]): number {
  let left = 0;
  let right = height.length - 1;
  let best = 0;
  while (left < right) {
    const area = (right - left) * Math.min(height[left], height[right]);
    best = Math.max(best, area);
    if (height[left] < height[right]) left++;
    else right--;
  }
  return best;
}
